shared/src/main/scala/org/atnos/eff/Interpret.scala

```diff
@@ -521,6 +521,15 @@ trait Interpret {
     }
   }
 
+  def augment[R, T[_], O[_], A](eff: Eff[R, A])(w: Augment[T, O])(implicit m: MemberInOut[T, R]): Eff[Fx.prepend[O, R], A] =  {
+    type U = Fx.prepend[O, R]
+    implicit val mw = MemberIn.MemberInAppendAnyL
+
+    translateInto(eff)(new Translate[T, U] {
+      def apply[X](tx: T[X]): Eff[U, X] = send[O, U, Unit](w(tx)) >> send[T, U, X](tx)
+    })
+  }
+
   def write[R, T[_], O, A](eff: Eff[R, A])(w: Write[T, O])(implicit m: MemberInOut[T, R]): Eff[Fx.prepend[Writer[O, ?], R], A] =  {
```

Comment:
Could you please define `write` in terms of `augment` since `augment` is a generalization?